persistence/sql/persister_identity.go

```diff
@@ -565,3 +598,103 @@ func (p *Persister) injectTraitsSchemaURL(ctx context.Context, i *identity.Ident
 	i.SchemaURL = s.SchemaURL(p.r.Config(ctx).SelfPublicURL()).String()
 	return nil
 }
+
+func (p *Persister) getJSONSearchQuery(ctx context.Context, field string, values []string) pop.ScopeFunc {
+	return func(q *pop.Query) *pop.Query {
+		// Removing this would allow SQL Injection. Never remove this.
+		if !validFields.MatchString(field) {
+			return q
+		}
+
+		prefix, key := extractFieldAndInnerFields(field)
+
+		switch p.Connection(ctx).Dialect.Name() {
+		case "sqlite3", "mysql", "mariadb":
+			for _, value := range values {
+				if key == "" {
+					q = q.Where(fmt.Sprintf(`json_extract(%s, '$') = ?`, prefix), value)
+				} else {
+					q = q.Where(fmt.Sprintf(`json_extract(%s, '$.%s') = ?`, prefix, key), value)
+				}
+			}
+			return q
+		case "postgres", "cockroach":
+			for _, value := range values {
+				param := fmt.Sprintf(`%s`, value)
+				if key != "" {
+					param = fmt.Sprintf(`{"%s":"%s"}`, key, value)
+				}
+				q = q.Where(fmt.Sprintf(`%s @> ?`, prefix), param)
+			}
+			return q
```

Comment:
@zepatrik what do you think about this? It's basically SQL injection, but with some sanization in front (see `!validFields.MatchString(field)`). I'm not 100% convinced though yet that we should merge it as is. Can you PTAL?